debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java

```diff
@@ -368,7 +382,21 @@ private Timer getTableScanLogTimer() {
     // TODO Should it be Statement or similar?
     // TODO Handle override option generically; a problem will be how to handle the dynamic part (Oracle's "... as of
     // scn xyz")
-    protected abstract String getSnapshotSelect(SnapshotContext snapshotContext, TableId tableId);
+    /**
+     * Generate a valid sqlserver query string for the specified table
+     *
+     * @param tableId the table to generate a query for
+     * @return a valid query string
+     */
+    private String getSnapshotSelect(TableId tableId) {
```

Comment:
This base class is shared between different implementations for SQL Server, Oracle etc, i.e. it cannot contain any specifics of SQL Server. For that reason it should remain an abstract method as before.